geode-redis/src/main/java/org/apache/geode/redis/internal/executor/EmptyRedisHash.java

```diff
@@ -26,41 +25,26 @@
 import org.apache.geode.redis.internal.executor.hash.RedisHash;
 
 public class EmptyRedisHash extends RedisHash {
-  @Override
-  public synchronized int hset(Region<ByteArrayWrapper, RedisData> region, ByteArrayWrapper key,
-      List<ByteArrayWrapper> fieldsToSet, boolean nx) {
-    throw new UnsupportedOperationException();
-  }
-
-  @Override
-  public synchronized int hdel(Region<ByteArrayWrapper, RedisData> region, ByteArrayWrapper key,
-      List<ByteArrayWrapper> fieldsToRemove) {
-    return 0;
-  }
-
-  @Override
-  public synchronized Collection<ByteArrayWrapper> hgetall() {
-    return emptyList();
+  public EmptyRedisHash() {
```

Comment:
Do we still need this class? I'm not sure I fully understand it's usage since when I switched `RedisHash.EMPTY` to `new RedisHash(Collections.emptyList())`, `HashesIntegrationTest` still passes.

Reply:
It is just an optimization. It gives us a single instance of RedisHash that represents an empty hash instead of having to create a new instance every time we want to represent an empty hash. We do this alot because we have a common pattern of not wanting to return null when we do the initial lookup of a RedisHash from the region.

Reply:
We do have to be careful that our canonical instance is immutable